### DISCUSSION TITLE
(중요) Member 구분자를 카카오 이메일로 하는 경우 관련 문제

### DISCUSSION THREAD

**Discussion by kmw2378** [KakaoFunding/back-end#24] [Q&A] [answered]
이전 카카오 로그인을 하면서 겪었던 문제가 이제 생각나서 글 올립니다ㅜㅜ

카카오 로그인의 경우 일단 사용자가 이메일 공개 여부를 동의하지 않으면 이메일 정보를 얻어올 수 없게 됩니다. 그래서 이전 프로젝트에서는 이메일 정보를 못얻어오면 임의로 이메일을 만들어 DB에 저장했습니다.
```java
public String getEmail() {
    final String email = (String) parsingProperties().get("email");
    if (email == null) {
        return getProviderId() + "@kakao.com";
    }

    return email;
}
```

근데 카카오쪽에서 이메일의 유일성이 보장되지 않는다고 하네요...?
<img width="813" alt="image" src="https://github.com/KakaoFunding/back-end/assets/107420002/b0bd650c-a252-4ed1-9b39-c0ceb4fe2d60">

어떻게 하는게 좋을까요? 저는 우선적으로 토큰에 이메일 대신 PK를 넣어야 될 거 같습니다..!

**Comment by hwan2-99** [comment]
이 부분 관련해서 저번에 다른프로젝트에서 토의한적이 있었는데, 그때도 저희는 pk를 사용하긴 했었습니다.

**Reply by kmw2378** [reply]
토큰에 pk를 사용했다는 말씀이실까요??

**Comment by YeaChan05** [comment]
카카오에서 `parsingProperties ()`로 사용자 속성값을 가져오는건가요?
그렇다면 pk말고 UUID값을 가져오는건 어떨까요?
unique한 값일거 같기도 하고 카카오에서 UUID를 제공해줬던거로 기억합니다

**Reply by kmw2378** [reply]
<img width="822" alt="image" src="https://github.com/KakaoFunding/back-end/assets/107420002/575045a8-dd39-48a8-8c6d-a591c4716eb4">

말씀하신 것 처럼 UUID를 사용하는 것도 좋을 것 같네요!

**Comment by kmw2378** [comment]
토큰에 UUID나 PK 중 1개를 넣어도 괜찮을 것 같네요! 

근데 `Member` 엔티티에 `email`이라는 항목이 필요할까요? 현재 이메일/비밀번호를 사용해서 로그인하질 않아서 둘 다 필요없어보이긴 합니다!

**Reply by YeaChan05** [reply, marked as answer]
저희 로직에 이메일이 굳이 필요할까라는 의문이 들긴 하네요
구현 내용에 따라 없어져도 무방하다고 생각합니다
